testutil/validatormock/validatormock.go

```diff
@@ -338,6 +338,43 @@ func ProposeBlindedBlock(ctx context.Context, eth2Cl Eth2Provider, signFunc Sign
 	return eth2Cl.SubmitBlindedBeaconBlock(ctx, signedBlock)
 }
 
+// this is register (singular) but the underlying submit function is registrations should it be changed to handle multiple registrations
+func Register(ctx context.Context, eth2Cl Eth2Provider, signFunc SignFunc, registration *eth2api.VersionedValidatorRegistration, pubkey eth2p0.BLSPubKey) error {
+	// TODO(corver): Sign and submit registration.
+
+	sigRoot, err := registration.V1.HashTreeRoot()
+	if err != nil {
+		return err
+	}
+
+	// epoch required for the GetDataRoot function not sure what we should set that as
+	sigData, err := signing.GetDataRoot(ctx, eth2Cl, signing.DomainApplicationBuilder, eth2p0.Epoch(100), sigRoot)
+	if err != nil {
+		return err
+	}
+
+	// pubkey required for signature
+	sig, err := signFunc(ctx, pubkey, sigData[:])
```

Comment:
if function is passed in multiple registrations and multiple pubkeys i'm guessing we would just loop through and use pubkey index i for registration index i?